src/runtime/classbase.cs

```diff
@@ -303,44 +301,40 @@ public static IntPtr tp_repr(IntPtr ob)
         public static void tp_dealloc(IntPtr ob)
         {
             ManagedType self = GetManagedObject(ob);
+            if (Runtime.PyType_SUPPORTS_WEAKREFS(Runtime.PyObject_TYPE(ob)))
```

Comment:
I rather split up the Runtime feature and PInvoke Calls. All interop calls as another file. As for the customized interop methods, use the partial class to split them, maybe? Since they all start with PyXXX_ and all from CPython, it would be convenient to use them by typing xxx.PyXXX_XXXX. PyXXX is enough to distinguish their category, disperse them may not good for use.

Reply:
@amos402 I like the idea. Can you move this to a class `CPythonMacros`? This would make it absolutely clear, that the method is a direct port of CPython code, and make it easy to find the original

Reply:
Another name of the class? Won't it better just use a partial class of Runtime into another file?

Reply:
Actually, you are probably right. It will work as is.

Ideally, we'd split `Runtime` into classes the same way they are grouped in CPython, so that would no longer be a single giant file. E.g. `Runtime.PyObject_Str` -> `PyObject.Str`, `Runtime.PyDict_GetItem` -> `PyDict.GetItem`, etc.